xsnap/sources/xsnap-worker.c

```diff
@@ -624,6 +628,20 @@ int main(int argc, char* argv[])
 						fprintf(stderr, "%s\n", fxWriteNetStringError(writeError));
 						c_exit(E_IO_ERROR);
 					}
+					// TODO reason through this follow-up from moddable's fork between 3.9.2 and 5.5.0
+					//path = nsbuf + 1;
+					//snapshot.stream = fopen(path, "rb");
+					//if (snapshot.stream) {
+					//	fxUseSnapshot(machine, &snapshot);
+					//	fclose(snapshot.stream);
+					//}
+					//else
+					//	snapshot.error = errno;
+					//if (snapshot.error) {
+					//	fprintf(stderr, "cannot restore snapshot %s: %s\n",
+					//			path, strerror(snapshot.error));
+					//	c_exit(E_IO_ERROR);
+					//}
```

Comment:
This section is present in the Moddable fork of `xsnap-worker.c`. It looks like it’s verifying that the snapshot can round-trip back. I found that some unit tests in `agoric-sdk/packages/xsnap` failed with this code in place. This may simply have been because the `path` was a file descriptor reference. If it’s useful, I could make this contingent on whether `path` starts with `@`.

attn @mhofman

Reply:
Ah yeah this was added by Moddable to reset the machine from snapshot to avoid performance degradation due to memory fragmentation. We do not need this as we do this upstream, and it is actually harmful to have that behavior hardcoded as we need the ability to test without reloading to detect load from snapshot divergence bugs.

Please take a look at discarded changes from https://github.com/agoric-labs/xsnap-pub/pull/44 compared to what was merged in https://github.com/agoric-labs/xsnap-pub/pull/46

Reply:
Also see https://github.com/agoric-labs/xsnap-pub/pull/48 for a proposal to make this use snapshot explicit.

We never landed it because we instead found it more efficient to start another worker and stream the snapshot to it (the agoric process is teeing the snapshot stream into both the swing-store and to the new worker).